models/QuizModel.php

```diff
@@ -887,4 +891,50 @@ public static function get_quiz_count_by_course( $course_id ) {
 		//phpcs:enable
 		return $count ? $count : 0;
 	}
+
+	/**
+	 * Get final quiz result depending on all attempts.
+	 *
+	 * @since 2.3.1
+	 *
+	 * @param int $quiz_id quiz id.
+	 * @param int $user_id user id.
+	 *
+	 * @return string pass, fail, pending
+	 */
+	public static function get_quiz_result( $quiz_id, $user_id = 0 ) {
+		global $wpdb;
+
+		$all_pending = true;
+		$result      = 'pending';
+
+		$user_id      = tutor_utils()->get_user_id( $user_id );
+		$attempt_list = $wpdb->get_results( $wpdb->prepare( "SELECT * FROM {$wpdb->prefix}tutor_quiz_attempts WHERE user_id=%d AND quiz_id=%d", $user_id, $quiz_id ) );
+
+		/**
+		 * Check all attempts are pending or not
+		 */
+		foreach ( $attempt_list as $attempt ) {
```

Comment:
Instead of looping to find out pending and calculate results, we can do it inside SQL. Here is an example:

`SELECT 
    attempt_id,
    user_id,
    quiz_id,
    total_marks,
    earned_marks,
    CAST((earned_marks / total_marks) * 100 AS UNSIGNED) AS percentage,
    (
        CASE 
            WHEN earned_marks >= (0.80 * total_marks) 
            THEN 'pass' 
            ELSE 'fail' 
        END
    ) AS result
FROM
    wp_tutor_quiz_attempts
WHERE
    attempt_status != 'review_required';
`

Reply:
With this query, we can't figure out whether quiz result (Single/Multiple) attempts are pending state or not.